frontend/libs/studio-components/src/hooks/useTableSorting.tsx

```diff
@@ -1,13 +1,53 @@
 import { useState, useEffect } from 'react';
 import type { Rows } from '../components';
+import { typedLocalStorage } from '@studio/pure-functions';
 
-export const useTableSorting = (rows: Rows, options: Record<'enable', boolean>) => {
-  const [sortColumn, setSortColumn] = useState(null);
-  const [sortDirection, setSortDirection] = useState('asc');
+export type SortPreference = {
+  column: string | null;
+  direction: 'asc' | 'desc';
+};
+
+export enum TableSortStorageKey {
+  Default = 'table-sort-preference',
+  OrgRepos = 'dashboard-org-repos-sort-preference',
+  FavoriteRepos = 'dashboard-favorite-repos-sort-preference',
+  DataModelRepos = 'dashboard-data-model-repos-sort-preference',
+}
+
+type TableSortingOptions = {
+  enable: boolean;
+  shouldPersistSort?: boolean;
+  storageKey?: TableSortStorageKey;
+};
+
+export const useTableSorting = (rows: Rows, options: TableSortingOptions) => {
+  const { enable, shouldPersistSort = false, storageKey = TableSortStorageKey.Default } = options;
+
+  const getSavedPreference = (): SortPreference | null => {
+    if (!shouldPersistSort) return null;
+    return typedLocalStorage.getItem<SortPreference>(storageKey);
+  };
+
+  const savedPreference = getSavedPreference();
+  const [sortColumn, setSortColumn] = useState<string | null>(savedPreference?.column ?? null);
+  const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>(
+    savedPreference?.direction ?? 'asc',
+  );
   const [sortedRows, setSortedRows] = useState<Rows>(rows);
 
+  const persistSortPreference = (column: string | null, direction: 'asc' | 'desc') => {
```

Comment:
Ref. prev comment we can do like following:
```suggestion
  const persistSortPreference = (column: string | null, direction: SortDirection) => {
```

Reply:
fixed